README.Rmd

```diff
@@ -0,0 +1,191 @@
+done
+
+# Restore
+cd "$working_directory"
+
+```
+
+For an easier way to work with all siblings at once from the terminal see
+[pacta-cli](https://github.com/2DegreesInvesting/pacta-cli).
+
+## Computing environment
+
+The required computing environment is available in [this docker
+image](https://github.com/2DegreesInvesting/docker/tree/master/r-packages).
+
+Other computing environments may be defined as workflows for GitHub actions.
+
+```{bash comment="$"}
+ls .github/workflows
+```
+
+Unless otherwise indicated, R packages should be up to date.
+
+<details> <summary>Session information.</summary>
+```{r}
+devtools::session_info()
+```
+</details>
+
+As an example, this is one way to use that computing environment to produce the
+very document you are now reading:
+
+* I move to the parent directory of PACTA_analysis and its siblings.
+
+```bash
+cd ~/git
+```
+
+* I run (`docker run`) an ephemeral (`--rm`), interactive (`-ti`) container of
+the image `2dii/r-packages`; create a number of volumes (`-v`) to access useful
+files in my host computer from inside the container; and set the working
+directory (`-w`) to PACTA_analysis/.
+
+```{r comment="", echo=FALSE}
+url <- "https://raw.githubusercontent.com/maurolepore/bin/master/docker-run-2dii-r-packages"
+tmp <- tempfile()
+download.file(url, tmp, quiet = TRUE)
+
+lines <- readLines(tmp)
+lines <- lines[2:length(lines)]
+writeLines(lines)
+```
+
+* Then I use R to produce the very file you are now reading.
+
+```r
+Rscript -e "rmarkdown::render('README.Rmd', output_format = 'md_document')"
+```
+
+## Installation
+
+To use the software in this and related repositories (more on this below) you
+need to work on developer mode. There is no "installation package"; instead
+you need to clone the source code [2DII's organization on
+GitHub](https://github.com/2DegreesInvesting/).
+
+Notice that this repository contains an R package but it is only for convenience
+and must also be used in developer mode.
+
+```{r}
+devtools::load_all()
+```
+
+## Reproducible examples
+
+This section shows reproducible examples of both the single- and multiple-inputs
+workflows. You may use these examples in at least three ways:
+
+* Read this document for an overview of which workflows are possible.
+* Run them on your local computer to track and explore their implementation.
+* Run them on a server as an integration test.
+
+### Single-inputs workflow
+
+The single-inputs workflow has three steps, each in its own script.
+
+```{bash comment="$"}
+ls web_tool_script*
+```
+
+With R, you can run each script individually with something like
+`source("web_tool_script_1.R")` or run multiple scripts at once. For example,
+you can run scripts 1 and 2 with:
+
+```{r}
+source_web_tool_scripts(1:2)
+```
+
+With the terminal, you can run each script individually. For example, you can
+run script 3 with:
+
+```{bash comment="$", error=TRUE}
+Rscript --vanilla web_tool_script_3.R TestPortfolio_Input  
+```
```

Comment:
will this work on Windows? I guess if someone has a bash shell emulator like git shell or something? I'm really not sure. Anyway, maybe make it more clear what is meant specifically by "terminal" and where that is or is not relevant?

Reply:
For now I'm commuting to only document a reproducible example in the computing environment that we maintain in the docker image. This document is actually rendered from inside that image.

Reply:
cool, so maybe change "With the terminal" to something like "From within the docker instance"?